src/ejabberd_c2s.erl

```diff
@@ -679,8 +679,7 @@ maybe_do_compress(El = #xmlel{name = Name, attrs = Attrs}, NextState, StateData)
     end.
 
 check_compression_auth(El, NextState, StateData) ->
```

Comment:
You've triggered meaningless changes. We prefer to match records in function clauses:

```erlang
check_compression_auth(El, NextState, StateData = #state{authenticated = false}) ->
    send_element_from_server_jid(StateData, compress_setup_failed()),
    fsm_next_state(NextState, StateData);
check_compression_auth(El, NextState, StateData) ->
    check_compression_method(El, NextState, StateData).
```

Reply:
Meh that was just my linter complaining about old code and I just changed it in place, I don't think it makes much of a difference but ok, I can do that xD